docs/introduction/create_custom_env.md

```diff
@@ -0,0 +1,222 @@
+```
+
+```python
+    def step(self, action):
+        # Map the action (element of {0,1,2,3}) to the direction we walk in
+        direction = self._action_to_direction[action]
+        # We use `np.clip` to make sure we don't leave the grid bounds
+        self._agent_location = np.clip(
+            self._agent_location + direction, 0, self.size - 1
+        )
+
+        # An environment is completed if and only if the agent has reached the target
+        terminated = np.array_equal(self._agent_location, self._target_location)
+        truncated = False
+        reward = 1 if terminated else 0  # the agent is only reached at the end of the episode
+        observation = self._get_obs()
+        info = self._get_info()
+
+        return observation, reward, terminated, truncated, info
+```
+
+## Registering and making the environment
+
+```{eval-rst}
+While it is possible to use your new custom environment now immediately, it is more common for environments to be initialized using :meth:`gymnasium.make`. In this section, we explain how to register a custom environment then initialize it.
+
+The environment ID consists of three components, two of which are  optional: an optional namespace (here: ``gymnasium_env``), a mandatory  name (here: ``GridWorld``) and an optional but recommended version (here: v0). It might have also been registered as ``GridWorld-v0`` (the  recommended approach), ``GridWorld`` or ``gymnasium_env/GridWorld``, and  the appropriate ID should then be used during environment creation.
+
+The entry point can be a string or function, as this tutorial isn't part of a python project, we cannot use a string but for most environment this is the normal way of specifying the entry point.
+
+Register has additionally parameters that can be used to specify keyword arguments to the environment, if to apply a time limit wrapper, etc. See :meth:`gymnasium.register` for more information.
+```
+
+```python
+gym.register(
+    id="gymnasium_env/GridWorld-v0",
+    entry_point=GridWorldEnv,
+)
+```
+
+For a more complete discussion on registering a custom environment (including with a string entry point), please read the full [create environment](../tutorials/gymnasium_basics/environment_creation) tutorial.
+
+```{eval-rst}
+Until the environment is registered, you can check with :meth:`gymnasium.pprint_registry`, then the environment can be initialized with :meth:`gymnasium.make` and even vectorized version with multiple instances of the same environment running in parallel with :meth:`gymnasium.make_vec`.
```

Comment:
Once the environment is registered, you can check the its indexc in the registry via :meth:`gymnasium.pprint_registry`, and the environment can then be initialized using :meth:`gymnasium.make`. A vectorized version of the environment with multiple instances of the same environment running in parallel can be instantiated with :meth:`gymnasium.make_vec`.